src/annotator/anchoring/test/range-js-test.js

```diff
@@ -51,81 +52,89 @@ describe('annotator/anchoring/range-js', () => {
-
-      $imports.$mock({
-        './xpath-util': {
-          findChild: fakeFindChild,
-        },
-      });
     });
 
-    afterEach(() => {
+    afterEach(function () {
       container.remove();
     });
 
-    it('returns the last node returned from `findChild`', () => {
-      const span = document.createElement('span');
-      container.appendChild(span);
-      fakeFindChild.onFirstCall().returns(container);
-      fakeFindChild.onSecondCall().returns(span);
-      assert.equal(nodeFromXPath('/div[1]/span[1]', document.body), span);
-    });
-
     [
       {
-        xpath: '/div[1]',
-        params: [
-          {
-            name: 'div',
-            idx: 1,
-          },
-        ],
+        xpath: '/h1[1]',
+        nodeName: 'H1',
       },
       {
-        xpath: '/div[1]/span[3]/p[1]',
-        params: [
-          {
-            name: 'div',
-            idx: 1,
-          },
-          {
-            name: 'span',
-            idx: 3,
-          },
-          {
-            name: 'p',
-            idx: 1,
-          },
-        ],
+        xpath: '/p[1]/a[1]/text()[1]',
+        nodeName: '#text',
       },
       {
-        xpath: '/DIV[2]/TEXT()[3]/SPAN[1]',
-        params: [
-          {
-            name: 'div',
-            idx: 2,
-          },
-          {
-            name: 'text()',
-            idx: 3,
-          },
-          {
-            name: 'span',
-            idx: 1,
-          },
-        ],
+        xpath: '/span[1]/ul[1]/li[2]',
+        nodeName: 'LI',
       },
     ].forEach(test => {
-      it('calls `findChild` with the following node names and indices', () => {
-        nodeFromXPath(test.xpath, document.body);
-        test.params.forEach(call => {
-          assert.calledWith(fakeFindChild, document.body, call.name, call.idx);
+      it('it returns the name name associated with the xpath', () => {
+        const result = nodeFromXPath(test.xpath, container);
+        assert.equal(result.nodeName, test.nodeName);
+      });
+    });
+
+    context('when `document.evaluate` throws an error', () => {
+      beforeEach(() => {
+        sinon.stub(document, 'evaluate').throws(new Error());
+        sinon.stub(console, 'log');
+      });
+
+      afterEach(() => {
+        // eslint-disable-next-line no-console
+        console.log.restore();
+        document.evaluate.restore();
+      });
+
+      [
+        {
+          xpath: '/h1[1]',
+          nodeName: 'H1',
+        },
+        {
+          xpath: '/p[1]/text()[1]',
+          nodeName: '#text',
+        },
+        {
+          xpath: '/p[2]/text()[2]',
+          nodeName: '#text',
+        },
+        {
+          xpath: '/SPAN[1]/UL[1]/LI[2]/text()',
+          nodeName: '#text',
+        },
+        {
+          xpath: '/SPAN[1]/UL[1]/LI[2]',
+          nodeName: 'LI',
+        },
+      ].forEach(test => {
+        it('it returns the name name associated with the xpath', () => {
```

Comment:
```suggestion
        it('it returns the node associated with the xpath', () => {
```

The function being tested returns a node.

As a suggestion, rather than comparing to a nodeName here, what you could do instead is check that the function returns the same result as `document.evaluate` would given the same XPath.